scanspec/service.py

```diff
@@ -62,12 +53,14 @@ class AxisFrames:
     """A fixed reference that can be scanned. i.e. a motor, time or
     number of repetitions.
     """
-    lower: Optional[Points]
-    """The lower bounds of each midpoint (used when fly scanning)"""
-    midpoints: Optional[Points]
-    """The centre points of the scan"""
-    upper: Optional[Points]
-    """The upper bounds of each midpoint (used when fly scanning)"""
+    lower: Points
+    """The lower bounds of each frame (used when fly scanning)"""
+    midpoints: Points
+    """The midpoints of each frame"""
+    upper: Points
+    """The upper bounds of each frame (used when fly scanning)"""
+    smallest_step: float
```

Comment:
Can we make this a resolver too?

Reply:
returned frames? if so then we can. 

If we wanted total frames as well then we would have to pass the path object before it has been affected by the max_points

Reply:
Sorry, I meant for `smallest_step`, that's the only thing that takes time to calculate